Added 'availability_template' to all Template Comp

grillp opened the pull request:
Added documentation `availability_template` configuration option for all Template platform components.

Update: Was PR: https://github.com/home-assistant/home-assistant/pull/25891
But split into 8 PRs as requested by @balloob here: https://github.com/home-assistant/home-assistant/pull/25891#issuecomment-528706857

https://github.com/home-assistant/home-assistant/pull/26509
https://github.com/home-assistant/home-assistant/pull/26510
https://github.com/home-assistant/home-assistant/pull/26511
https://github.com/home-assistant/home-assistant/pull/26512
https://github.com/home-assistant/home-assistant/pull/26513
https://github.com/home-assistant/home-assistant/pull/26514
https://github.com/home-assistant/home-assistant/pull/26516
https://github.com/home-assistant/home-assistant/pull/26517

## Checklist:

- [X] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [X] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html


<a href="https://gitpod.io/#https://github.com/home-assistant/home-assistant.io/pull/10123"><img src="https://gitpod.io/api/apps/github/pbs/github.com/grillp/home-assistant.io.git/3245873adf520e285b99380743ca83959fc1c5f8.svg" /></a>

grillp committed:
Added 'availability_template' to all Template Comp

grillp committed:
Mised some indenting and a `

grillp committed:
Merge branch 'next' into next

frenck commented:
You should have just 1 PR for each parent PR. The problem with the approach you are currently using, is that if 4 PR's get merged, but the others don't... then we can't merge the docs?
Please split the documentation PR to match the parent PR's.

grillp commented:
OK @frenck,

I split it into 8 PRs and updated the code pull requests to reference the corresponding .io PR:

https://github.com/home-assistant/home-assistant.io/pull/10307
https://github.com/home-assistant/home-assistant.io/pull/10308
https://github.com/home-assistant/home-assistant.io/pull/10309
https://github.com/home-assistant/home-assistant.io/pull/10310
https://github.com/home-assistant/home-assistant.io/pull/10311
https://github.com/home-assistant/home-assistant.io/pull/10312
https://github.com/home-assistant/home-assistant.io/pull/10313
https://github.com/home-assistant/home-assistant.io/pull/10314
